### PR TITLE
HIPP-273: New data model in frontend

### PR DESCRIPTION
**Pull request by PaulCDurham**
https://jira.tools.tax.service.gov.uk/browse/HIPP-273

### CODE REVIEW

**Review by tigdik — APPROVED** [review]

**Comment by laurahk** [comment]
Not sure these are related to this PR but noticed a few warnings when I ran `sbt clean test`:

`warning: lazy value publishingSettings in object SbtDistributablesPlugin is deprecated (since 1.0.2): publishingSettings is not needed
  .settings(SbtDistributablesPlugin.publishingSettings: _*)`

and 5 unused import warnings for `ModelGenerators.scala` and `UserAnswersEntryGenerators.scala`

File: app/models/application/NewScope.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.application
+
+import play.api.libs.json.{Format, Json}
+
+case class NewScope(name: String, environments: Seq[EnvironmentName])
```
Comment (laurahk) [inline app/models/application/NewScope.scala:21]: This makes me think we are making a new scope. Perhaps `NewScopeRequest` might be clearer?

File: app/models/application/ScopeStatus.scala [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+case object Pending extends WithName("PENDING") with ScopeStatus
+case object Approved extends WithName("APPROVED") with ScopeStatus
+case object Denied extends WithName("DENIED") with ScopeStatus
+
+object ScopeStatus extends Enumerable.Implicits {
+
+  val values: Seq[ScopeStatus] = Seq(Pending, Approved, Denied)
+
+  implicit val enumerable: Enumerable[ScopeStatus] =
+    Enumerable(values.map(value => value.toString -> value): _*)
```
Comment (laurahk) [inline app/models/application/ScopeStatus.scala:32]: This seems quite boilerplate so perhaps not, but should it have a unit test?

File: app/controllers/CreateApplicationController.scala [hunk 3]
```diff
@@ -42,15 +43,15 @@ class CreateApplicationController @Inject()(
...
+          .map(app => Redirect(routes.CreateApplicationSuccessController.onPageLoad(app.id)))
       )
   }
 
-  private def validateAndBuildApplication(userAnswers: UserAnswers): Either[Call, Application] = {
+  private def validateAndBuildApplication(userAnswers: UserAnswers): Either[Call, NewApplication] = {
     for {
       applicationName <- validateApplicationName(userAnswers)
-    } yield Application(None, applicationName)
+    } yield NewApplication(applicationName, Creator(email = ""))
```
Comment (laurahk) [inline app/controllers/CreateApplicationController.scala:54]: There's a small chance we won't send the creator with the request after discussing at tomorrow's meeting about authenticating frontend/backend. But I doubt it, and it's easily changed :)

**Review by laurahk — APPROVED** [review]
Looks fab! So neat 😄

**Pull request merged**